### PR TITLE
feat:support c-style escape string 

### PR DESCRIPTION
**Pull request by ZENOTME**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

This PR support c-style escape string in risingwave. [More detail](https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-SYNTAX-STRINGS-UESCAPE:~:text=4.1.2.2.%C2%A0String%20Constants%20With%20C%2DStyle%20Escapes) about c-style escape in postgres
```
// postgres

test_db=> select e'\u003f';
 ?column?
----------
 ?
(1 row)

//RisingWave

dev=> select e'\u003f';
ERROR:  QueryError: Item not found: Invalid column: e
```

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
#6733

### CODE REVIEW

File: src/sqlparser/tests/sqlparser_common.rs [hunk 1]
```diff
@@ -1968,9 +1968,9 @@ fn parse_literal_decimal() {
 
 #[test]
 fn parse_literal_string() {
-    let sql = "SELECT 'one', N'national string', X'deadBEEF'";
+    let sql = "SELECT 'one', N'national string', X'deadBEEF', E'c style escape string'";
```
Comment (xiangjinwu) [inline src/sqlparser/tests/sqlparser_common.rs:1971]: Although we have e2e sqllogictest, it is still better to make this test/example a little bit more representative.

File: src/sqlparser/src/tokenizer.rs [hunk 2]
```diff
@@ -373,6 +376,21 @@ impl<'a> Tokenizer<'a> {
                         }
                     }
                 }
+                x @ 'e' | x @ 'E' => {
+                    chars.next(); // consume, to check the next char
+                    match chars.peek() {
+                        Some('\'') => {
+                            // E'...' - a <character string literal>
+                            let s = self.tokenize_single_quoted_string(chars)?;
```
Comment (xiangjinwu) [inline src/sqlparser/src/tokenizer.rs:384]: There is a subtle difference between single quoted string and escaped string:
> Also, a single quote can be included in an escape string by writing `\'`, in addition to the normal way of `''`.

In PostgreSQL:
```
test=# select 'a\'z;
 z  
----
 a\
(1 row)

test=# select e'a\'z';                                                                                                                                       ?column? 
----------
 a'z
(1 row)
```

**Review by xiangjinwu — COMMENTED** [review]
rest lgtm

**Review by xxchan — COMMENTED** [review]
Just curious: Why unescape is in binder instead of parser? 🤔

**Comment by ZENOTME** [comment]
> Just curious: Why unescape is in binder instead of parser? 🤔
> 
> Oh, it can make unparse simpler

It's just to keep parse simpler.🤣

File: e2e_test/batch/basic/escape_string.slt.part [hunk 3]
```diff
@@ -0,0 +1,35 @@
+query I
+select e'\ta'
+----
+    a
+
```
[the remark this answers is not in the document]
Comment (xiangjinwu) [reply e2e_test/batch/basic/escape_string.slt.part:5]: Let's add these 2 as test cases.

File: src/sqlparser/src/ast/value.rs [hunk 4]
```diff
@@ -64,6 +65,7 @@ impl fmt::Display for Value {
             Value::SingleQuotedString(v) => write!(f, "'{}'", escape_single_quote_string(v)),
             Value::NationalStringLiteral(v) => write!(f, "N'{}'", v),
             Value::HexStringLiteral(v) => write!(f, "X'{}'", v),
+            Value::CstyleEscapesString(v) => write!(f, "E'{}'", v),
```
Comment (xiangjinwu) [inline src/sqlparser/src/ast/value.rs:68]: Just though of another issue 🤦, about unparsing.

Try to create a new yaml file under `src/sqlparser/tests/testdata`:
```
- input: SELECT 'a''b'
  formatted_sql: SELECT 'a''b'

- input: SELECT e'a''b'
  formatted_sql: SELECT E'a''b'
```

The actual unparse result would produce an invalid query: `SELECT E'a'b'`.

Note that unparsing does not need to reproduce the original query, but it should produce an equivalent one. To be more specific, it is okay to unparse `e'a\'b''c'` into `e'a''b''c'` or `e'a\'b\'c'`.

No matter which of the 3 styles above you choose, make sure to test the unparsing of `\\` as well. For example:
```
select e'\\', e'\'', e'\\\'', e'\\\'''', e'\z';
```

Comment (xiangjinwu) [reply src/sqlparser/src/ast/value.rs:68]: Maybe the easiest fix is to push `\'` when `tokenize_single_quoted_string_with_escape` sees `''`.

Rationale:
* The `SingleQuotedString` converts `''` to `'` during parsing and reverts it during unparsing.
* The `CstyleEscapesString` attempts to preserve escape sequences until binding, to make unparsing easy. But `''` is being treated the same way as in `SingleQuotedString` during parsing.

**Review by xiangjinwu — APPROVED** [review]

**Pull request merged**